Include.md

```diff
@@ -0,0 +1,144 @@
+# Modules
+
+## Summary
+
+We propose adding a module extension mechanism to the WESL shading language.
+
+## Assumptions
+
+Assumes that [`load`](./Imports.md), [Modules](./Modules.md) and [Module Interfaces](./ModulesInterfaces.md) have already been implemented. The last one however isn't essential to this proposal which could be tweaked to function without it.
+
+# Motivation
+
+While WESL currently provides a way to organize and encapsulate types, functions and state. However it does *not* 
+provide a way to specialize behaviour, or in other words does not provide a mechanism for inheritance. 
+
+Specialization is important for a number cases, an important and frequent one of which is for abstracting the implementation and configuration details of Physically Based Rendering (PBR), reducing the boilerplate and number of edge cases a technical artist has to consider. 
+
+At its core, PBR is a mathematical parametrisation of the interaction of an infintesimally small fragment of a physical materials with light. For example many implementations of PBR allow one to specify the physical qualities of a surface such as the albedo, normal, how rough or smooth a surface is and whether it is metallic or not. These qualities are used in a complex integral that is used to drive the appearance of a surface. However most workflows do not require
+technical artists to know the details of these integrals, and instead allow them to author materials by simply varying 
+the PBR values across the surface. 
+
+An example of how we would want such our abstraction to look like in a game engine such as [Bevy](https://bevyengine.org/)
+would be as follows:
+
+```reasonml
+// Including standard material adds all the symbols 
+// to the global module including the vertex and fragment entry points. 
+include BevyPbr::StandardMaterial;
+
+// Tells the linker to resolve calls to `resolve_pbr_inputs`
+// to this function instead of the base. This is constrained to the 
+// current namespace; in this case the namespace is the global scope.
+patch fn resolve_pbr_inputs(pbr_inputs: &PbrInputs) {
```

Comment:
Patch seems powerful! too powerful? 
Are patch targets constrained somehow? (I'm thinking about the library case, where a library author wouldn't want messy internals to become public api because a user can patch them.)

Reply:
The visibility mechanism in this proposal would be the only way to constrain patch targets which is quite limiting I agree. An alternative idea would be something like a "virtual" keyword.